packages/editor/src/components/rich-text/index.js

```diff
@@ -220,7 +221,7 @@ export class RichText extends Component {
 	 * @return {Object} The current record (value and selection).
 	 */
 	getRecord() {
-		const { formats, text } = this.formatToValue( this.props.value );
+		const { formats, text } = this.formatToValue( this.props.value, this.props.annotations );
```

Comment:
#11595 is merged we can use this API now :). Let me know if you need help refactoring.